stdlib/public/runtime/EnvironmentVariables.cpp

```diff
@@ -0,0 +1,165 @@
+//===--- CompatibiltyOverride.h - Debug variables. --------------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// Debug behavior conditionally enabled using environment variables.
+//
+//===----------------------------------------------------------------------===//
+
+#include "swift/Runtime/Debug.h"
+#include "swift/Runtime/EnvironmentVariables.h"
+
+#include <errno.h>
+
+using namespace swift;
+
+// Require all environment variable names to start with SWIFT_
+static constexpr bool hasSwiftPrefix(const char *str) {
+  const char prefix[] = "SWIFT_";
+  for (unsigned i = 0; i < sizeof(prefix) - 1; i++)
```

Comment:
Is there a reason to prefer the open-coded `memcmp` ?  There is a hard-enough dependency on the C library that I think that we can get away with it.

Reply:
I wasn't able to come up with any nicer `constexpr` way to do it. `memcmp` and friends aren't `constexpr`. I fiddled with `__builtin_constant_p` but that just produces `false` for calls to functions like this.

Reply:
FTR, there are concerns over `constexpr`-ness of `memcmp`, so it has been open-coded here.